### DISCUSSION TITLE
Silly question about comments in code template

### DISCUSSION THREAD

**Discussion by rutledj** [nextgenhealthcare/connect#6288] [Q&A] [answered]
How can I get my comments in a code template to show up correctly (line breaks retained) when I hit ctrl space for function info?

My code template looks like this:

![image](https://github.com/user-attachments/assets/853f32b1-290b-4934-96d6-00cd09ed0adc)

but when I try to use it and view the comments, it is all run together

![image](https://github.com/user-attachments/assets/922a75a6-6921-4a6b-99cb-f925ed36c882)

**Comment by pacmano1** [comment]
<img width="176" alt="image" src="https://github.com/user-attachments/assets/b8472eb1-abbe-4c2b-9e9b-a19041cc5846">

Click that on the lower right and follow the format it throws in your code template

**Comment by rutledj** [comment]
Unfortunately, that didn't change anything.

**Reply by pacmano1** [reply]
Go over to the sample code templates section and copy what they do there.   You will have problems with trying to add line feeds, so rework your stuff as needed.  // note @odoodo comment, I was wrong on line feeds.

**Comment by odoodo** [comment, marked as answer]
You can use html w/i the JavascriptDoc comments. A linebreak would e.g. be generated by a &lt;br/&gt; tag.

**Reply by pacmano1** [reply]
Kinda fun to mess around with this

```JS
/**
	This function does stuff and stuff and some more stuff and yet some more stuff.  
	<br/>
     <br/> 
     Sample call: msg = <b>removeUnwantedMRNs</b>(msg,{'{EPIC','BNMG')

	@param {String} root - (<i>required</i>) The HL7 Message
	@param {String} MRNsToKeep - (<i>required</i>) An Array of assigning authorities you want to keep
	@return {Object} No clue 
*/

function removeUnwantedMRNs(root, MRNsToKeep) {
  // TODO: Enter code here
}
```
![image](https://github.com/user-attachments/assets/126b01a3-1564-4086-9fa2-c4216db62a42)

**Reply by odoodo** [reply]
Indeed. And it can really add some value. As soon as I get some time for it I will create a Javascript Doc generater. It will allow to generate documentation for Mirth code templates & channels. (similar to Javadoc but w/ some additional information like dependencies)
